feat (anticheat\db\conf): Skyfire Anticheat Manager

acidmanifesto opened the pull request:
Detections for the following cheats\hacks:
-   *speed
-   *fly
-   *jump
-   *waterwalk
-   *teleportplane
-   *climb
-   *time maniputation
-   *gravity
-   *teleportrts
-   *bg start

Counter Measures for the following cheats\hacks:
-   *speed
-   *fly
-   *jump
-   *waterwalk
-   *time maniputation
-   *teleportrts
*BG Start should have a core counter already so one is not offered in this. if play violates the start spot of applicable BG's if the match has not started.

Automatic Moderation Features (Fan request but not really recommended...):
- Auto Kick
- Auto Jail
- Auto Ban

New sql tables introduced:
Character DB:

- daily_players_reports
- players_reports_status

New Skyfire String:

- 11002

New GM Cmds:

- .anticheat handle (off\on) 
turns it  off\on anticheat. 
Example .anticheat handle off 

- .anticheat global
Gives you a global average of cheat hits with the highest and lowest violators

- .anticheat player PLAYERNAME  or select the player
give you a print off of current cheat stats 
Example .anticheat player dumbshit  or select the player and type .anticheat

- .anticheat delete PLAYERNAME   or select the player
Deletes all cheat records of the player for the current session
EXAMPLE .anticheat delete dumbshit  or select the player and type .anticheat

- .anticheat purge
Deletes all cheat records for everyone's current session.

- .anticheat jail PLAYERNAME  or select the player
Jails the player and sets his hearthstone to GM Jail with debuffs to keep them from teleporting or queing anything.
Example .anticheat jail DUMBASS  or select the player and type .anticheat jail

- .anticheat parole PLAYERNAME  or select the player
Paroles the player from GM Jail and teles them to faction capital and removes the debuffs and sets their hearth stones to the faction capital
EXAMPLE .anticheat parole DUMBASS or select the player and type .anticheat parole

Prefer Conf Settings
```

###################################################################################################
#   ANTICHEAT
#
#     Anticheat.Enable
#       Description: Enables or disables the Anticheat System functionality
#       Default:     1 - (Enabled)
#                    0 - (Disabled)

Anticheat.Enable = 1

#     Anticheat.EnabledOnGmAccounts
#       Description: Enables detection for GM accounts
#       Default:    0 - (Disabled)
#                   1 - (Enabled)

Anticheat.EnabledOnGmAccounts = 1

#      Anticheat REPORTING
#       Description: Alerts GMs that are on line once reports hit designated threshhold and stops after desiganted threshhold
#       Default:    70 - (Alerts at and Stops Alerting at 70 report thresh hold)
#                    0 - (Disabled)
Anticheat.Reports.InGame = 70
Anticheat.Reports.InGame.Max = 70

#     Anticheat.Detect
#       Description: It represents which detections are enabled (ClimbHack disabled by default).
#       Default:    0 - (Disabled)
#                   1 - (Enabled)

Anticheat.DetectFlyHack = 1
Anticheat.DetectWaterWalkHack = 1
Anticheat.DetectJumpHack = 1
Anticheat.StricterDetectJumpHack = 1
Anticheat.DetectTelePlaneHack = 1
Anticheat.DetectSpeedHack = 1
Anticheat.DetectClimbHack = 1
Anticheat.DetectGravityHack = 1
Anticheat.DetectTelePortHack = 1

#   ANTICHEAT COUNTER MEASURE
#
#     Anticheat Counter Measures
#
#     Anticheat.CM.Teleport
#
#       Description: Sends player back to last gps position if player cheats and teleports.
#
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#
Anticheat.CM.Teleport = 1
#
#     Anticheat.CM.FLYHACK
#
#       Description: Sets player back to the ground if fly hack is detected
#
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#
Anticheat.CM.FLYHACK = 1
#
#     Anticheat.CM.WATERHACK
#
#       Description: Sets player to fall if water hack is detected
#
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#
Anticheat.CM.WATERHACK = 1
#
#     Anticheat.CM.JUMPHACK
#     Anticheat.CM.ADVJUMPHACK
#       Description: Sets player back to the ground if jump hack is detected
#
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#
Anticheat.CM.JUMPHACK = 1
Anticheat.CM.ADVJUMPHACK = 1
#
#     Anticheat.CM.SPEEDHACK
#
#       Description: Sets player back to allowed server rate speed when speed hack is detected
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#
Anticheat.CM.SPEEDHACK = 1

#     Automatic Moderation Features
#
#     Anticheat.KickPlayer
#     Anticheat.ReportsForKick
#
#       Description: Enables and Auto kick when reports reach threshhold
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#       Default:    70 - (Kicks at 70 auto reports)
#

Anticheat.KickPlayer = 0
Anticheat.ReportsForKick = 75

#     Anticheat.BanPlayer
#     Anticheat.ReportsForBan
#
#       Description: Enables and Auto ban when reports reach threshhold
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#       Default:    70 - (Bans at 70 auto reports)
#

Anticheat.BanPlayer = 0
Anticheat.ReportsForBan = 70

#     Anticheat.JailPlayer
#     Anticheat.ReportsForJail
#
#       Description: Enables and Auto Jail when reports reach threshhold
#       Default:    0 - (Disabled)
#                   1 - (Enabled)
#       Default:    70 - (Jails at 70 auto reports)
#

Anticheat.JailPlayer = 0
Anticheat.ReportsForJail = 70

# Anticheat Logging\Appenders
# Currently spams into its own file
# If you want world console spam just use Logger.anticheat=3,Console Server Anticheat

Appender.Anticheat=2,3,15,anticheat.log,a
Logger.anticheat=3,Server Anticheat

#
###################################################################################################

```

acidmanifesto committed:
feat (anticheat): Anticheat Port

acidmanifesto committed:
update (anticheat): Remove Lua Blocker

acidmanifesto committed:
update (anticheat): Ace Singleton

acidmanifesto committed:
update (anticheat): singleton pt 2

acidmanifesto committed:
update (anticheat): LUA Blocker removal pt2

acidmanifesto committed:
update (anticheat): Remove unused variable

acidmanifesto committed:
update (anticheat): Lua Blocker Removal PT 3

AriDEV3 commented on src/server/game/Anticheat/AnticheatMgr.h:
```diff
@@ -0,0 +1,123 @@
+    ~AnticheatMgr();
+
+    public:
+        static AnticheatMgr *instance()
```
friend class ACE_Singleton<ObjectMgr, ACE_Null_Mutex>;

acidmanifesto replied:
thank you.
48a1d9749a

acidmanifesto committed:
update (anticheat): Move ace singleton

acidmanifesto committed:
update (anticheat): Remove lua blocker leftover

acidmanifesto committed:
update (anticheat): Reduce code complexity PT1

acidmanifesto committed:
chore (antcheat): whitespace

acidmanifesto committed:
update (anticheat): include cleanup

acidmanifesto committed:
update (anticehat): static clean up

acidmanifesto committed:
update (anticheat): Revert define change

acidmanifesto committed:
update (anticheat)L cmake cmd fix

acidmanifesto committed:
update (anticheat): Parole gps correction

acidmanifesto committed:
update (anticheat): Added case default

acidmanifesto committed:
update (anticheat): Twin Peaks BG

acidmanifesto committed:
update (anticheat): Replace case swith a better if

SkyFire commented:
Still testing.

acidmanifesto commented:
> Still testing.

Just fyi. Still trying to figure out a rework for those code factor issues. Its just taking a bit longer then expected.

acidmanifesto committed:
update (anticheat): False hits for select elevators

acidmanifesto committed:
add (anticheat\log): add logging

this is to address race condition when mapupdate threads are set higher then 2 in the wold conf

acidmanifesto committed:
update (anticheat): Reorder

this works better for the race condition

acidmanifesto committed:
update (anticheat): :Reorder

SkyFire commented:
Fix this already :)
